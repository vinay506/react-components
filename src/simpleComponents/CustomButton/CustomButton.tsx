import React from 'react'
import { Button } from '@rmwc/button';
import '@rmwc/button/styles';
export const BUTTON_TYPES = {
    RAISED: 'RAISED',
    UNELEVATED: 'UNELEVATED',
    OUTLINED: 'OUTLINED',
    DENSE: 'DENSE'
}
function CustomButton(props: any) {
    /**
     * Text: The content to be displayed on the button
     * type: Constants provided in BUTTON_TYPES can be used
     * onButtonClick: is a call back function when user clicks on the button
     * color: Text color on the button
     * bgColor: Background color of the button
     * borderColor: Border color of the button 
     */
    const { text, type, onButtonClick, color, bgColor, borderColor,width,height,padding,margin } = props;
    const buttonStyle = {
        color: color,
        backgroundColor: bgColor,
        borderColor: borderColor,
        width:width,
        height:height,
        padding:padding,
        margin:margin
    };

    let button;
    switch (type) {
        case BUTTON_TYPES.RAISED:
            button = <><Button id="raised" raised label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
        case BUTTON_TYPES.DENSE:
            button = <><Button id="dense" dense label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
        case BUTTON_TYPES.OUTLINED:
            button = <><Button id="outlined" outlined label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
        case BUTTON_TYPES.UNELEVATED:
            button = <><Button id="unelevated" unelevated label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break
        default:
            button = <><Button raised label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
    }
    return (
        <React.Fragment>
            {button}
        </React.Fragment>
    )
}

export default CustomButton;


