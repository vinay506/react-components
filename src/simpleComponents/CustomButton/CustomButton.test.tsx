import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import CustomButton, { BUTTON_TYPES } from './CustomButton';

describe('Custom Button Outlined', () => {
    let wrapper: any;
    let propInfo: any;
    const buttonEvent = () => {
        return true;
    }
    beforeEach(() => {
        propInfo = {
            text: 'Get Started',
            type: BUTTON_TYPES.OUTLINED,
            onButtonClick: buttonEvent,
            color: 'blue',
            bgColor: 'white',
            borderColor: 'blue'
        }

        wrapper = shallow(<CustomButton {...propInfo} />);
    });
    test('Custom Button Snapshot', () => {
        const instance = renderer.create(<CustomButton {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Custom Button Element check', () => {
        const elementLength = wrapper.find('#outlined').length;
        expect(elementLength).toEqual(1);
    })
    test('Custom Button Styles Validation', () => {
        const elementStyles = wrapper.find('#outlined');
        expect(elementStyles.props().style).toEqual({
            color: propInfo.color,
            backgroundColor: propInfo.bgColor,
            borderColor: propInfo.borderColor
        });
    })

    test('Custom Button Click Test', () => {

        const wrapper = shallow(<CustomButton onButtonClick={buttonEvent} type={BUTTON_TYPES.OUTLINED} />);
        const button = wrapper.find('#outlined');
        const clickEvent = button.simulate('click');
        const returnValue = expect(clickEvent).toBeCalled;
        expect(returnValue).toBeTruthy();
    })
})