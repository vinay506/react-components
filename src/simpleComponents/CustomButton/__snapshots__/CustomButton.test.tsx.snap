// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button Outlined Custom Button Snapshot 1`] = `
<button
  className="mdc-button mdc-button--outlined"
  id="outlined"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "blue",
      "color": "blue",
    }
  }
>
  <div
    className="mdc-button__ripple"
  />
  <span
    className="mdc-button__label"
  >
    Get Started
  </span>
</button>
`;
